File: frontend/console/src/api/department.js
import Axios from "@/http";

export function createDepartment(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/department', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveDepartment(departmentId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/department/${ departmentId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateDepartment(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/department', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteDepartment(departmentId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/department/${ departmentId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteDepartmentWithFlag(departmentId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/department/${ departmentId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllDepartments() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/department/all').then(res => {
            resolve(res)
        })
    })
}

export function queryDepartment (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/department/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/department.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_DEPARTMENT = "/config/department"
const URL_ROUTER_DEPARTMENT_BY_ID = "/config/department/:department-id"
const URL_ROUTER_DEPARTMENT_QUERY = "/config/department/query"
const URL_ROUTER_DEPARTMENT_ALL = "/config/department/all"

func DepartmentRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_DEPARTMENT, config.CreateDepartment)
	r.GET(URL_ROUTER_DEPARTMENT_BY_ID, config.RetrieveDepartment)
	r.PUT(URL_ROUTER_DEPARTMENT, config.UpdateDepartment)
	r.DELETE(URL_ROUTER_DEPARTMENT_BY_ID, config.DeleteDepartment)
	r.PUT(URL_ROUTER_DEPARTMENT_BY_ID, config.DeleteDepartmentWithFlag)
	r.POST(URL_ROUTER_DEPARTMENT_QUERY, config.QueryDepartment)
	r.GET(URL_ROUTER_DEPARTMENT_ALL, config.GetAllDepartments)
	return r
}